django_cas_ng/backends.py

```diff
@@ -88,6 +88,13 @@ def authenticate(self, request: HttpRequest, ticket: str, service: str) -> Optio
         if pgtiou and settings.CAS_PROXY_CALLBACK and request:
             request.session['pgtiou'] = pgtiou
 
+        # Map CAS affiliations to Django groups
+        if settings.CAS_MAP_AFFILIATIONS and user and attributes:
+            affils = attributes.get('affiliation')
+            for affil in affils:
```

Comment:
error handling, what about there is no `affiliation`. It may crash.

Reply:
Good point! I updated line 93, to default to an empty array if there is no `affiliation`. That should resolve the issue, but let me know other suggestions you have.

By the way, I intended to add tests for this change, but had trouble getting them to work.

Reply:
Good point! I updated line 93, to default to an empty array if there is no `affiliation`. That should resolve the issue, but let me know other suggestions you have.

By the way, I intended to add tests for this change, but had trouble getting them to work.

Reply:
Test would be the best example to show how it works.
Hope https://github.com/django-cas-ng/django-cas-ng/blob/master/docs/test.rst can help you write your test?